tests/src/python/test_qgsgeometry.py

```diff
@@ -2867,6 +2867,85 @@ def testConvertToMultiType(self):
         assert compareWkt(expWkt, wkt), "testConvertToMultiType failed: mismatch Expected:\n%s\nGot:\n%s\n" % (
             expWkt, wkt)
 
+    def testConvertToCurvedMultiType(self):
+        """ Test converting geometries to multi curve type """
+        point = QgsGeometry.fromWkt('Point (1 2)')
+        assert point.convertToCurvedMultiType()
+        expWkt = 'MultiPoint ((1 2))'
+        wkt = point.asWkt()
+        assert compareWkt(expWkt, wkt), "testConvertToCurvedMultiType failed: mismatch Expected:\n%s\nGot:\n%s\n" % (
+            expWkt, wkt)
+        # test conversion of MultiPoint
+        assert point.convertToCurvedMultiType()
+        assert compareWkt(expWkt, wkt), "testConvertToCurvedMultiType failed: mismatch Expected:\n%s\nGot:\n%s\n" % (
+            expWkt, wkt)
+
+        multipoint = QgsGeometry.fromWkt('MultiPoint ((1 2))')
+        assert multipoint.convertToCurvedMultiType()
+        expWkt = 'MultiPoint ((1 2))'
+        wkt = multipoint.asWkt()
+        assert compareWkt(expWkt, wkt), "testConvertToCurvedMultiType failed: mismatch Expected:\n%s\nGot:\n%s\n" % (
+            expWkt, wkt)
+        # test conversion of MultiPoint
+        assert multipoint.convertToCurvedMultiType()
+        assert compareWkt(expWkt, wkt), "testConvertToCurvedMultiType failed: mismatch Expected:\n%s\nGot:\n%s\n" % (
+            expWkt, wkt)
+
+        line = QgsGeometry.fromWkt('LineString (1 0, 2 0)')
+        assert line.convertToCurvedMultiType()
+        expWkt = 'MultiCurve (LineString (1 0, 2 0))'
+        wkt = line.asWkt()
+        assert compareWkt(expWkt, wkt), "testConvertToCurvedMultiType failed: mismatch Expected:\n%s\nGot:\n%s\n" % (
+            expWkt, wkt)
+        # test conversion of MultiCurve
+        assert line.convertToCurvedMultiType()
+        assert compareWkt(expWkt, wkt), "testConvertToCurvedMultiType failed: mismatch Expected:\n%s\nGot:\n%s\n" % (
+            expWkt, wkt)
+
+        circular = QgsGeometry.fromWkt('CircularString (0 0, 1 1, 2 0)')
+        assert circular.convertToCurvedMultiType()
+        expWkt = 'MultiCurve (CircularString (0 0, 1 1, 2 0))'
+        wkt = circular.asWkt()
+        assert compareWkt(expWkt, wkt), "testConvertToCurvedMultiType failed: mismatch Expected:\n%s\nGot:\n%s\n" % (
+            expWkt, wkt)
+        # test conversion of MultiCurve
+        assert circular.convertToCurvedMultiType()
+        assert compareWkt(expWkt, wkt), "testConvertToCurvedMultiType failed: mismatch Expected:\n%s\nGot:\n%s\n" % (
+            expWkt, wkt)
+
+        multiline = QgsGeometry.fromWkt('MultiLineString ((1 0, 2 0))')
+        assert multiline.convertToCurvedMultiType()
+        expWkt = 'MultiCurve (LineString (1 0, 2 0))'
```

Comment:
should this be MultiCurve( CompoundCurve( ) ) ? That's what I would have immediately expected from this function.

Reply:
a LineString is a legal component in a MultiCurve, since it is a Curve.

Reply:
I realise that -- it's just that the method name "convert to curved ... type" immediately suggests to me that the result would be an explicit curved geometry type. (Like if we had a non-promote-to-multi version of this I'd expect feeding a single linestring would result in a compound curve output.)

Reply:
the use case here is that the top "geometry" must be a curved multi geometry type. e.g to be able to export a WFS layer mixing linestring, compoundcurve, multilinestring, etc. as a multicurve one for formats like GeoPackage or PostGIS that strictly enforce the "top" geometry type. The inner type doesn't matter much, as long as it is legal for the container.

Reply:
not necessarily, the ultimate source of truth, but the behaviour I implemented is similar to OGR ForceTo(wkbMultiCurve):
```
>>> g = ogr.CreateGeometryFromWkt('LINESTRING(0 0,1 1)')
>>> g = ogr.ForceTo(g, ogr.wkbMultiCurve)
>>> g.ExportToIsoWkt()
'MULTICURVE ((0 0,1 1))'
>>> g.GetGeometryRef(0).ExportToIsoWkt()
'LINESTRING (0 0,1 1)'
```